### PR TITLE
smlnj: 110.95 -> 110.99.3

### PR DESCRIPTION
**Pull request by pi314mm**
###### Description of changes

updated smlnj to 110.99.3

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
LGTM, please squash commits. Also change PR and commit title format to `smlnj: 110.95 -> 110.99.3`

File: pkgs/development/compilers/smlnj/default.nix [hunk 1]
```diff
@@ -48,7 +48,7 @@ in stdenv.mkDerivation {
   inherit sources;
 
   patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/smlnj/default.nix:50]:
```suggestion
  postPatch = ''
```

File: pkgs/development/compilers/smlnj/default.nix [hunk 2]
```diff
@@ -1,45 +1,45 @@
 { lib, stdenv, fetchurl, Libsystem }:
 let
-  version = "110.95";
+  version = "110.99.3";
   baseurl = "http://smlnj.cs.uchicago.edu/dist/working/${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/smlnj/default.nix:4]:
```suggestion
  baseurl = "https://smlnj.org/dist/working/${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by lvitaly** [comment]
By the way, this update of the `smlnj` should bring apple silicon support under the Rosetta2 translator (see [110.99.1](https://www.smlnj.org/dist/working/110.99.1/HISTORY.html)) as well as the support of macOS 13 (see [110.99.3](https://www.smlnj.org/dist/working/110.99.3/110.99.3-README.html)). It would be great to know that apple silicon will be covered by nix as well 🙏

**Comment by onny** [comment]
Please squash commits :)

**Comment by SuperSandro2000** [comment]
darwin fails to build with `` ../../config/gen-posix-names.sh: line 85: xcrun: command not found``

**Pull request closed without merging**